### ISSUE TITLE
Tests in lazyLoading engine doesn't work

### ISSUE THREAD

**Issue by mydea** [ember-engines/ember-engines#341]
I can't seem to get any type of test to run in a lazy engine.

My `test-helper.js` file looks like this:

```js
import resolver from './helpers/resolver';
import {
  setResolver
} from 'ember-qunit';
import preloadAssets from 'ember-asset-loader/test-support/preload-assets';
import manifest from 'dummy/config/asset-manifest';

preloadAssets(manifest);
setResolver(resolver);
```

But then I get a bunch of errors, and none of the tests are even run:

```
Uncaught TypeError: this.doneCallbacks.pop(...) is not a function
    at Class.asyncEnd (adapter.js:14)
...
Uncaught TypeError: Cannot read property 'assert' of undefined
    at Class.asyncStart (adapter.js:10)
```

My main dependencies look like this:

```json
{
  "dependencies": {
    "ember-cli-babel": "~5.2.0",
    "ember-cli-htmlbars": "~1.1.1",
    "ember-cli-shims": "~1.0.2",
    "ember-data": "~2.11.0",
    "ember-source": "~2.11.0"
  },
  "devDependencies": {
    "broccoli-asset-rev": "~2.5.0",
    "ember-cli": "2.11.0",
    "ember-cli-dependency-checker": "~1.3.0",
    "ember-cli-htmlbars-inline-precompile": "~0.3.3",
    "ember-cli-qunit": "~3.1.0",
    "ember-cli-uglify": "~1.2.0",
    "ember-engines": "0.4.0",
    "ember-load-initializers": "~0.6.3",
    "ember-resolver": "~2.1.1",
    "loader.js": "~4.1.0"
  },
  "peerDependencies": {
    "ember-engines": "0.4.0"
  }
}
```

When I set `lazyLoading: false` in my index.js, everything works as expected. 
Is there anything else I need to do? Or do tests simply not work for lazy engines at the moment at all?

**Comment by trentmwillis** [comment]
So this issue is related to https://github.com/emberjs/ember-qunit/issues/246. Basically, with QUnit 2.x you can't control tests async starting/stopping outside a test (which makes sense), but this leads to issues because _all_ RSVP Promises use `asyncStart` and `asyncStop` on the testing adapter, regardless of if they're in a test or not.

I'm unsure of the proper fix since I haven't had time to dig in. So if you can mess around with it and hopefully come up with a path forward, that would be helpful!

**Comment by mydea** [comment]
I forked ember-qunit and played around a bit and got it working with my app.

I changed `adapter.js` a bit: 
https://github.com/mydea/ember-qunit/blob/fix-async-behaviour-outside-tests/lib/ember-qunit/adapter.js

Not sure how "pretty" this change is and if it might have negative side effects somewhere? But the tests work as expected for me now.

If that would work for you, I'll make a PR, or make other changes if someone can point me in the right direction.
